File: 2016/finale/labigriglia/att/labigriglia.pas
const
    MAXN = 1000;
type
    matrix = array[0..MAXN-1] of array[0..MAXN-1] of longint;

(* Mettete qui le eventuali variabili globali *)

function cammina(N, M: longint; var V: matrix): longint;
begin
    (* Mettete qui il codice della soluzione *)
    cammina := 42;
end;


var
    N, M, i, j : longint;
    griglia    : array[0..MAXN-1] of array[0..MAXN-1] of longint;
    fr, fw: text;

begin
    assign(fr, 'input.txt');
    assign(fw, 'output.txt');
    reset(fr);
    rewrite(fw);
    readln(fr, N, M);
    for i:=0 to N-1 do
        for j:=0 to M-1 do
            read(fr, griglia[i, j]);

    writeln(fw, cammina(N, M, griglia));
    close(fr);
    close(fw);
end.

File: ed2/finale/labigriglia/att/labigriglia.c
#include <stdio.h>
#include <assert.h>

#define MAXN 1000

int cammina(int N, int M, int griglia[][MAXN]) {
    // Mettete qui il codice della soluzione
    return 42;
}


int griglia[MAXN][MAXN];

int main() {
    FILE *fr, *fw;
    int N, M, i, j;

    fr = fopen("input.txt", "r");
    fw = fopen("output.txt", "w");
    assert(2 == fscanf(fr, "%d%d", &N, &M));
    for(i=0; i<N; i++)
        for(j=0; j<M; j++)
            assert(1 == fscanf(fr, "%d", &griglia[i][j]));

    fprintf(fw, "%d\n", cammina(N, M, griglia));
    fclose(fr);
    fclose(fw);
    return 0;
}
